examples/model_compress/auto_pruners_torch.py

```diff
@@ -263,129 +298,140 @@ def evaluator(model):
+            scheduler = MultiStepLR(
+                optimizer, milestones=[int(args.fine_tune_epochs*0.5), int(args.fine_tune_epochs*0.75)], gamma=0.1)
+        elif args.dataset == 'cifar10' and args.model == 'resnet18':
+            optimizer = torch.optim.SGD(model.parameters(), lr=0.1, momentum=0.9, weight_decay=5e-4)
+            scheduler = MultiStepLR(
+                optimizer, milestones=[int(args.fine_tune_epochs*0.5), int(args.fine_tune_epochs*0.75)], gamma=0.1)
+        elif args.dataset == 'cifar10' and args.model == 'resnet50':
+            optimizer = torch.optim.SGD(model.parameters(), lr=0.1, momentum=0.9, weight_decay=5e-4)
+            scheduler = MultiStepLR(
+                optimizer, milestones=[int(args.fine_tune_epochs*0.5), int(args.fine_tune_epochs*0.75)], gamma=0.1)
+        best_acc = 0
+        for epoch in range(args.fine_tune_epochs):
+            train(args, model, device, train_loader, criterion, optimizer, epoch)
+            scheduler.step()
+            acc = evaluator(model)
+            if acc > best_acc:
+                best_acc = acc
+                torch.save(model.state_dict(), os.path.join(args.experiment_data_dir, 'model_fine_tuned.pth'))
 
-    if args.save_model:
-        pruner.export_model(os.path.join(
-            args.experiment_data_dir, 'model_fine_tuned.pth'), os.path.join(args.experiment_data_dir, 'mask.pth'))
-        print('Fined tuned model saved to %s', args.experiment_data_dir)
+    print('Evaluation result (fine tuned): %s' % best_acc)
+    print('Fined tuned model saved to %s', args.experiment_data_dir)
+    result['performance']['finetuned'] = best_acc
 
-    # model speed up
-    if args.speed_up and args.pruner != 'AutoCompressPruner':
-        if args.model == 'LeNet':
-            model = LeNet().to(device)
-        elif args.model == 'vgg16':
-            model = VGG(depth=16).to(device)
-        elif args.model == 'resnet18':
-            model = models.resnet18(pretrained=False, num_classes=10).to(device)
-        elif args.model == 'mobilenet_v2':
-            model = models.mobilenet_v2(pretrained=False).to(device)
-
-        model.load_state_dict(torch.load(os.path.join(args.experiment_data_dir, 'model_fine_tuned.pth')))
-        masks_file = os.path.join(args.experiment_data_dir, 'mask.pth')
-
-        m_speedup = ModelSpeedup(model, dummy_input, masks_file, device)
-        m_speedup.speedup_model()
-        evaluation_result = evaluator(model)
-        print('Evaluation result (speed up model): %s' % evaluation_result)
-        result['speedup'] = evaluation_result
-
-        torch.save(model.state_dict(), os.path.join(args.experiment_data_dir, 'model_speed_up.pth'))
-        print('Speed up model saved to %s', args.experiment_data_dir)
-
-    with open(os.path.join(args.experiment_data_dir, 'performance.json'), 'w+') as f:
+    with open(os.path.join(args.experiment_data_dir, 'result.json'), 'w+') as f:
```

Comment:
Can we save all results into one json?  maybe use a key to identify different experiment?

Reply:
> Can we save all results into one json? maybe use a key to identify different experiment?

Merged to 3 files. One file by every dataset/model combination.